### PR TITLE
New Speak Command connected to Virtual Alexa Device 

### LINKED ISSUE #359: Updated speak command
We want to integrate the Silent Echo into our bst command-line.

To do this, we need to:
1) Re-brand the existing speak command that uses the emulator as "utter"
2) Create a new speak command that uses silent echo

For new the command, it should be defined as follows:
bst speak [options] <complete utterance>

The options are:
>--token The token for interacting with your virtual Alexa device - get it here:
>      https://apps.bespoken.io/dashboard/validation
> (Bespoken Dashboard account required for use of this feature)

The token is saved off to the .bst/config file when entered, so it does not need to be re-entered every time.

The phrase and the token should be used to call the silent-echo-sdk. The result should include:
* The transcript of the output (if present)
* The stream audio URL (if present)
* The card data (if present)

Suggested layout:
> Transcript:
> Welcome to we study billionaires!
>
> Stream:
> https://mystream.com/link.mp3
>
> Card:
> My Card Title
> My Card Subtitle
> My Card Content
> https://mycard.com/image_url

The existing "speak" command should behave the same, though the name should be changed to utter.

### PR DESCRIPTION
**Pull request by jperata**
(Closes #359)

This PR is not yet ready to merge, putting it out there for revision on the logic.

What it has:
 - Rename old speak command to utter.
 - Add handler for silent echo sdk.
 - Create new speak command connecting with silent echo as specified on #359 

What is missing:
 - ~Update documentation for utter and speak.~
 - ~Tests for new speak command.~
 - ~Current implementation returns the "The Silent Echo service is no longer available." mesage.~
 - ~We need #368 to be merged first into master and then into this branch to solve test failures.~
 - ~Update to a url different than https://silentecho-dev.bespoken.io/process when available~

### CODE REVIEW

File: bin/bst-speak.ts [hunk 1]
```diff
@@ -30,62 +27,50 @@ program
...
+        if (token) {
+            console.log("Your token is saved, you can now use this command without providing a token");
+        }
 
-        speaker.start(function (error: string) {
-            if (error !== undefined) {
-                process.exit(0);
-                return;
-            }
+        console.log("Transcript:");
```
Comment (jkelvie) [inline bin/bst-speak.ts]: I would move the logic for calling SilentEcho and formatting the output to its own class. For two reasons:
1) The logic is a bit more complex, with lots of ifs (as almost anything can be null)
2) We will keep adding to it, with render templates, video, etc.
These additional may even require us creating HTML output, so the user can view it.

For that reason, having this handling in its own class makes sense

File: bin/bst-speak.ts [hunk 2]
```diff
@@ -30,62 +27,50 @@ program
...
+        console.log(silentEchoResponse.transcript);
```
Comment (jkelvie) [inline bin/bst-speak.ts]: This can be null, so requires an if check

File: bin/bst-speak.ts [hunk 3]
```diff
@@ -30,62 +27,50 @@ program
...
+        console.log();
+        console.log("Stream:");
+        console.log(silentEchoResponse.transcriptAudioURL);
```
Comment (jkelvie) [inline bin/bst-speak.ts]: This should be the streamURL, not the transcriptAudioURL.

It can also be null

File: bin/bst-speak.ts [hunk 4]
```diff
@@ -30,62 +27,50 @@ program
...
+        console.log();
+        if (silentEchoResponse.card) {
+            console.log("Card:");
+            silentEchoResponse.card.mainTitle && console.log(silentEchoResponse.card.mainTitle);
```
Comment (jkelvie) [inline bin/bst-speak.ts]: These fields can all be null, except for textField (that one should always be set if there is a card)

File: bin/bst-speak.ts [hunk 5]
```diff
@@ -1,20 +1,17 @@
...
-    .option("-u, --url <alexa-skill-url>", "The URL of the Alexa skill to speak to - defaults to current proxied skill")
-    .option("-i, --intents <intent-schema-path>", "Path to the intent schema file - defaults to ./speechAssets/IntentSchema.json")
-    .option("-s, --samples <sample-utterances-path>", "Path to the sample utterances file - defaults to ./speechAssets/SampleUtterances.txt")
-    .option("-a, --appId <application-id>", "The application ID for the skill")
-    .option("-U, --userId <user-id>", "Sets the user id to the specified value")
-    .option("-t, --accessToken <accessToken>", "Sets the access token for emulating a user with a linked account")
-    .description("Creates an intent request based on the specified utterance and sends it to your skill")
-    .action(function () {
+    .option("-t, --token <token>", `The token for interacting with your virtual Alexa device - get it here:\n" +
+        "https://apps.bespoken.io/dashboard/validation\n" +
```
Comment (jkelvie) [inline bin/bst-speak.ts]: This urls needs to be /dashboard/skills/<SKILL_ID>/validation

File: bin/bst-speak.ts [hunk 6]
```diff
@@ -30,62 +27,35 @@ program
...
-                console.error("No URL specified and no proxy is currently running");
+        let silentEchoResponse;
+        try {
+            silentEchoResponse = await SilentEchoClient.speak(utterance, token);
+        } catch (error) {
+            if (error.message === "Token Required") {
+                console.log("You need a token for this option to work, get it here:");
                 console.log();
-                console.log("URL (--url) must be specified if no proxy is currently running");
+                console.log("\thttps://apps.bespoken.io/dashboard/skills/" + Global.config().sourceID() + "/validation");
```
Comment (jkelvie) [inline bin/bst-speak.ts]: Same as above

File: docs/index.md [hunk 7]
```diff
@@ -42,7 +42,8 @@ The command-line tools include:
 
 * [Deploy](http://docs.bespoken.io/en/latest/commands/deploy) - One-command deployment of Lambda functions
 * [Proxy](http://docs.bespoken.io/en/latest/commands/proxy) - Easily debug Alexa requests by having them send directly to your laptop
-* [Speak](http://docs.bespoken.io/en/latest/commands/speak) and [Intend](http://docs.bespoken.io/en/latest/commands/intend) - Send intents and utterances to your Alexa skill
+* [Speak](http://docs.bespoken.io/en/latest/commands/speak) - Communicate to a virtual device to test your skills with text as if you were voice commands.
```
Comment (jkelvie) [inline docs/index.md]: "as if you were" -> "as if they were"

File: docs/commands/utter.md [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+It works in a manner very similar to the Alexa simulator available via the Alexa developer console.  
+
+To start using it, you will need a local file that contains your Intent Schema and Sample Utterances.  
+By default, we have adopted the pattern used by the Alexa Skills Sample projects (see [here](https://github.com/alexa/skill-sample-nodejs-hello-world/)).
+
+That is, we look for the Interaction Model files inside a folder called speechAssets located off the source root.  
+
+You can specify an alternative location via options to the command-line.
+
+## Speaking
```
Comment (jkelvie) [inline docs/commands/utter.md]: "Speaking" -> "Uttering"

File: docs/commands/speak.md [hunk 9]
```diff
@@ -1,16 +1,9 @@
 
 ## Overview
 
-The speak command generates intent requests for your service as if they were coming from Alexa itself.
+The speak command communicates to a virtual device to test your skills with text as if you were voice commands.
```
Comment (jkelvie) [inline docs/commands/speak.md]: "as if you were" -> "as if they were"

File: bin/bst-speak.ts (hunk 6 above)
Comment (jperata) [reply bin/bst-speak.ts]: you mean we shouldn't autocomplete the skill id?

**Pull request merged**